core/unmarshal_v2_primitives_test.go

```diff
@@ -0,0 +1,1386 @@
+	jsonString = strings.ReplaceAll(jsonString, "%f1", fmt.Sprintf("%f", f1))
+
+	rawMap := unmarshalMap(jsonString)
+	assert.NotNil(t, rawMap)
+
+	model := new(MyModel)
+
+	var err error
+
+	// Positive tests
+	err = UnmarshalPrimitive(rawMap, "prop", &model.Prop)
+	assert.Nil(t, err)
+	assert.NotNil(t, model.Prop)
+	assert.Equal(t, s1, model.Prop.(string))
+
+	err = UnmarshalPrimitive(rawMap, "prop_slice", &model.PropSlice)
+	assert.Nil(t, err)
+	assert.NotNil(t, model.PropSlice)
+	assert.Equal(t, n1, int64(model.PropSlice[0].(float64)))
+	assert.Equal(t, n2, int64(model.PropSlice[1].(float64)))
+
+	err = UnmarshalPrimitive(rawMap, "prop_map", &model.PropMap)
+	assert.Nil(t, err)
+	assert.NotNil(t, model.PropMap)
+	assert.Equal(t, b1, model.PropMap["key1"].(bool))
+	assert.Equal(t, b2, model.PropMap["key2"].(bool))
+
+	err = UnmarshalPrimitive(rawMap, "prop_slice_map", &model.PropSliceMap)
+	assert.Nil(t, err)
+	assert.NotNil(t, model.PropSliceMap)
+	assert.Equal(t, n1, int64(model.PropSliceMap["key1"][0].(float64)))
+	assert.Equal(t, n2, int64(model.PropSliceMap["key1"][1].(float64)))
+	assert.Equal(t, n2, int64(model.PropSliceMap["key2"][0].(float64)))
+	assert.Equal(t, n1, int64(model.PropSliceMap["key2"][1].(float64)))
+
+	err = UnmarshalPrimitive(rawMap, "prop_map_slice", &model.PropMapSlice)
+	assert.Nil(t, err)
+	assert.NotNil(t, model.PropMapSlice)
+	assert.Equal(t, f1, model.PropMapSlice[0]["key1"].(float64))
+	assert.Equal(t, f1, model.PropMapSlice[1]["key2"].(float64))
+
+	err = UnmarshalPrimitive(rawMap, "prop_slice_map_slice", &model.PropSliceMapSlice)
+	assert.Nil(t, err)
+	assert.NotNil(t, model.PropSliceMapSlice)
+	assert.Equal(t, s1, model.PropSliceMapSlice[0]["key1"][0].(string))
+	assert.Equal(t, n1, int64(model.PropSliceMapSlice[1]["key2"][0].(float64)))
+	assert.Equal(t, n1, int64(model.PropSliceMapSlice[1]["key2"][1].(float64)))
+	assert.Equal(t, n2, int64(model.PropSliceMapSlice[1]["key2"][2].(float64)))
+
+	// Tests involving a JSON null value
+	model.Prop = "bad value"
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.Prop)
+	assert.Nil(t, err)
+	assert.Nil(t, model.Prop)
+
+	model.PropSlice = []interface{}{"bad1", "bad2"}
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropSlice)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropSlice)
+
+	model.PropMap = map[string]interface{}{ "key1": "value1" }
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropMap)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropMap)
+
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropSliceMap)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropSliceMap)
+
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropMapSlice)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropMapSlice)
+
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropSliceMapSlice)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropSliceMapSlice)
+
+	// Negative tests
+	model.Prop = nil
+	err = UnmarshalPrimitive(rawMap, "bad_type", &model.Prop)
```

Comment:
This seems wrong to me.  Type "Any" should allow any type -- how can we have a bad type?

Reply:
Come on Mike, I was using alliteration in my testcase development :)  lol

Reply:
hahahaha